lib/symengine/basic.rb

```diff
@@ -11,8 +11,13 @@ def free_symbols
     end
     
     def abs
-        return SymEngine::abs(self)
+      SymEngine::abs(self)
+    end
+    def to_proc()
+      if self.free_symbols.count > 1
+        raise ArgumentError, "Too many free symbols! Only 1 allowed. Found #{self.free_symbols.count}."
```

Comment:
That's also possible. It's basically a matter of better user experience. I didn't want to make users to type in something like `[[1, 2], [3,4]].map(&func.to_proc([x, y]))` for two argument functions while for 1 argument functions it was `[1, 2].map(&func)`.

imho, it would be simpler if the user just creates a lambda, (`lam = SymEngine::lambdify(func, [x, y])`) and calls the lambda from there onwards. `[[1, 2], [3, 4]].map(&lam)` 

This looks much clear.

But maybe for those who are more familiar with the ever so complex ruby code, it may be better to allow both :D Let me know what you think, I'll make `to_proc` work like you described.